### PR TITLE
CIV-9198 - finalise trial arrangements - hearing duration & other info

### PR DESCRIPTION
**Pull request by m-meulendijks-v1**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9198

### Change description ###
Additions were also made to previous pages to allow for saving along the way.

Page for trial duration & other information input fields.
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — APPROVED** [review]

File: src/main/modules/i18n/locales/cy.json [hunk 1]
```diff
@@ -123,6 +124,17 @@
       "SIGN_IN": "Mewngofnodi",
       "SIGN_OUT": "Allgofnodi"
     },
+    "HEARING_DURATION": {
+      "MINUTES_30": "30 minutes",
+      "MINUTES_60": "1 hour",
+      "MINUTES_90": "1 and a half hours",
+      "MINUTES_120": "2 hour",
+      "MINUTES_150": "2 and a half hour",
```
Comment (hmcts-version1-ignacio) [inline src/main/modules/i18n/locales/cy.json:132]: "MINUTES_120": "2 hour", 
 "MINUTES_150": "2 and a half hour",
 
 Missing "s"?

File: src/main/modules/i18n/locales/en.json [hunk 2]
```diff
@@ -123,6 +124,17 @@
       "SIGN_IN": "Sign in",
       "SIGN_OUT": "Sign out"
     },
+    "HEARING_DURATION": {
+      "MINUTES_30": "30 minutes",
+      "MINUTES_60": "1 hour",
+      "MINUTES_90": "1 and a half hours",
+      "MINUTES_120": "2 hour",
+      "MINUTES_150": "2 and a half hour",
```
Comment (hmcts-version1-ignacio) [inline src/main/modules/i18n/locales/en.json:132]: "MINUTES_120": "2 hour",
"MINUTES_150": "2 and a half hour",

Missing "s"?

File: src/main/modules/i18n/locales/cy.json (hunk 1 above)
Comment (m-meulendijks-v1) [reply src/main/modules/i18n/locales/cy.json]: Added the 's' - well spotted!

File: src/main/modules/i18n/locales/en.json (hunk 2 above)
Comment (m-meulendijks-v1) [reply src/main/modules/i18n/locales/en.json]: Added the 's' - well spotted!

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by kurt-uk — DISMISSED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by miguelMolina3691 — APPROVED** [review]

File: src/main/common/models/caseProgression/HearingDuration.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  MINUTES_150 = 'MINUTES_150',
+  MINUTES_180 = 'MINUTES_180',
+  MINUTES_240 = 'MINUTES_240',
+  DAY_1 = 'DAY_1',
+  DAY_2 = 'DAY_2',
+}
+
+export function formatHearingDuration (hearingDuration: HearingDuration)
+{
+  return t('COMMON.HEARING_DURATION.'+hearingDuration.toString());
```
Comment (LeonardoPalmeiro) [inline src/main/common/models/caseProgression/HearingDuration.ts:17]: please format this file

File: src/main/common/models/caseProgression/caseProgressionHearing.ts [hunk 4]
```diff
@@ -44,4 +47,7 @@ export class CaseProgressionHearing {
     return HearingDateTimeFormatter.getHearingDateFormatted(this.hearingDate, lang);
   }
 
+  getHearingDurationFormatted(): string {
+    return formatHearingDuration(this.hearingDuration);
```
Comment (LeonardoPalmeiro) [inline src/main/common/models/caseProgression/caseProgressionHearing.ts:51]: can we use the same approach of line 47?
image

File: src/main/routes/features/caseProgression/trialArrangements/hearingDurationAndOtherInformationController.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+const hearingDurationViewPath = 'features/caseProgression/trialArrangements/hearing-duration-other-info';
+const hearingDurationController = Router();
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+hearingDurationController.get([TRIAL_ARRANGEMENTS_HEARING_DURATION], (async (req, res, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const claim = await civilServiceClient.retrieveClaimDetails(claimId, <AppRequest>req);
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/caseProgression/trialArrangements/hearingDurationAndOtherInformationController.ts:29]: I'm not surewe should to use retrieveClaimDetails or const claim = await getClaimById(claimId, req);

File: src/main/routes/features/caseProgression/trialArrangements/hearingDurationAndOtherInformationController.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  }
+}) as RequestHandler);
+
+hearingDurationController.post([TRIAL_ARRANGEMENTS_HEARING_DURATION], (async (req, res, next) => {
+  try {
+    //TODO: save trial information using the commented out information below
+    //const otherInfo = req.body.otherInformation;
+    //const form = new OtherTrialInformation(otherInfo);
+
+    //TODO: check your answers URL will have to be checked as part of CIV-9201
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/caseProgression/trialArrangements/hearingDurationAndOtherInformationController.ts:48]: I'm not sure why we are not save the information on redis.
should we save or not?

File: src/main/common/models/caseProgression/HearingDuration.ts (hunk 3 above)
Comment (m-meulendijks-v1) [reply src/main/common/models/caseProgression/HearingDuration.ts]: Formatted

File: src/main/common/models/caseProgression/caseProgressionHearing.ts (hunk 4 above)
Comment (m-meulendijks-v1) [reply src/main/common/models/caseProgression/caseProgressionHearing.ts]: I have added the HearingDurationFormatter file to get a similar style.

File: src/main/routes/features/caseProgression/trialArrangements/hearingDurationAndOtherInformationController.ts (hunk 5 above)
Comment (m-meulendijks-v1) [reply src/main/routes/features/caseProgression/trialArrangements/hearingDurationAndOtherInformationController.ts]: I've switched to await getClaimById

File: src/main/common/models/caseProgression/trialArrangements/TrialArrangements.ts
[the remark this answers is not in the document]
Comment (m-meulendijks-v1) [reply src/main/common/models/caseProgression/trialArrangements/TrialArrangements.ts:1]: renamed

File: src/main/common/models/caseProgression/trialArrangements/trialArrangements.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import {HasAnythingChangedForm} from 'models/caseProgression/trialArrangements/hasAnythingChangedForm';
+import {YesNo} from 'form/models/yesNo';
+
+export class TrialArrangements{
+  isCaseReady?: YesNo;
```
Comment (LeonardoPalmeiro) [inline src/main/common/models/caseProgression/trialArrangements/trialArrangements.ts:5]: I believe in this part is   isCaseReady?: IsCaseReadyForm;

File: src/main/common/utils/pageSectionBuilder.ts
Comment (LeonardoPalmeiro) [inline src/main/common/utils/pageSectionBuilder.ts:1]: I'm not sure why we're change the methods of position. do we really want to do that?

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]
please add a test to getHearingDurationFormatted method.

File: src/main/common/form/models/caseProgression/trialArrangements/OtherTrialInformation.ts
[the remark this answers is not in the document]
Comment (m-meulendijks-v1) [reply src/main/common/form/models/caseProgression/trialArrangements/OtherTrialInformation.ts:1]: Changed to camelCase

File: src/main/common/models/caseProgression/HearingDuration.ts
[the remark this answers is not in the document]
Comment (m-meulendijks-v1) [reply src/main/common/models/caseProgression/HearingDuration.ts:1]: Changed to camelCase

File: src/main/common/models/caseProgression/trialArrangements/trialArrangements.ts (hunk 7 above)
Comment (m-meulendijks-v1) [reply src/main/common/models/caseProgression/trialArrangements/trialArrangements.ts:5]: IsCaseReadyForm just contains a YesNo, so I retrieve that YesNo when saving to TrialArrangements.
I use the form only when sending/retrieving things from the view.

File: src/main/common/utils/pageSectionBuilder.ts
Comment (m-meulendijks-v1) [reply src/main/common/utils/pageSectionBuilder.ts:1]: I think it makes more sense to have mainTitle & leadParagraph before title & paragraph, or not?
I think I added them in, and then someone else added them in a different place as part of their PR, and their PR got merged first.

File: src/main/common/models/caseProgression/trialArrangements/finaliseYourTrialSectionBuilder.ts [hunk 8]
```diff
@@ -55,6 +55,30 @@ export class FinaliseYourTrialSectionBuilder extends UploadYourDocumentsSectionB
...
+      data: {
+        html: '<p class="govuk-body">'+text+'</p>',
+        variables: variables,
+      },
+    });
+    this._claimSummarySections.push(paragraphSection);
+    return this;
+  }
+
+  addInsetText(text: string, variables?: unknown) {
```
[the remark this answers is not in the document]
Comment (m-meulendijks-v1) [reply src/main/common/models/caseProgression/trialArrangements/finaliseYourTrialSectionBuilder.ts]: I have moved it to pageSectionBuilder - I think that was a good idea.

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Pull request merged**